### ISSUE TITLE
waiting for jobs when default partition is not interactive

### ISSUE THREAD

**Issue by ChristopherBarrington** [SESYNC-ci/rslurm#64]
In the `wait_for_job` function, `srun` is used to hold R execution until the dependency is completed. This failed for me because our default partition does not accept interactive jobs. My solution was to use `sbatch` to wrap `localhost` instead (below). I also had problems when `wait_for_job` was used inside an `sbatch`ed script - the `srun` was run without waiting for the dependency (I think using the resources of the parent job). Both seem to have been resolved using sbatch instead of srun in `wait_for_job`.

https://github.com/SESYNC-ci/rslurm/blob/c32e7b90c2365ee6002c02140db4485a7c5e7639/R/slurm_utils.R#L52

```r
# Submit dummy job with a dependency via sbatch to block R process
wait_for_job <- function(slr_job) {
    queued <- system(
        paste('test -z "$(squeue -hn', slr_job$jobname, '2>/dev/null)"'), ignore.stderr = TRUE)
    if (queued) {
        sbatch <- sprintf('sbatch --nodes=1 --output=/dev/null --time=00:01:00 --dependency=singleton --job-name=%s --wait --wrap="hostname"', slr_job$jobname)
        system(sbatch, intern=TRUE)
    }
    return()
}
```

**Comment by qdread** [comment]
Hello and thanks for the issue. rslurm was developed for a HPC that accepts interactive jobs across all partitions using `srun`, so it's useful to know that `wait_for_job` breaks if that isn't the case. Would you be interested in submitting a pull request with the modified version of the function that works in both cases?

**Comment by ChristopherBarrington** [comment]
I forked your repository this morning and made a change from srun to sbatch in the function. Tested on our cluster and looks good. I have created a pull request with that one change. Thanks.

**Comment by qdread** [comment]
Resolved by #65 . thanks again!
